.github/workflows/check-nixf-tidy.yml

```
@@ -0,0 +1,97 @@
+          baseRev=$(git rev-parse HEAD^1)
+          git worktree add "$base" "$baseRev"
+          echo "baseRev=$baseRev" >> "$GITHUB_ENV"
+          echo "base=$base" >> "$GITHUB_ENV"
+      - name: Get Nixpkgs revision for nixf
+        run: |
+          # pin to a commit from nixpkgs-unstable to avoid e.g. building nixf
+          # from staging
+          # This should not be a URL, because it would allow PRs to run arbitrary code in CI!
+          rev=$(jq -r .rev ci/pinned-nixpkgs.json)
+          echo "url=https://github.com/NixOS/nixpkgs/archive/$rev.tar.gz" >> "$GITHUB_ENV"
+      - uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b # v27
+        with:
+          # explicitly enable sandbox
+          extra_nix_config: sandbox = true
+          nix_path: nixpkgs=${{ env.url }}
+      - name: Install nixf and jq
+        # provided jq is incompatible with our expression
+        run: "nix-env -f '<nixpkgs>' -iAP nixf jq"
+      - name: Check that Nix files pass nixf-tidy
+        run: |
+          failedFiles=()
+
+          # TODO: Make this more parallel
+
+          # Loop through all Nix files touched by the PR
+          while readarray -d '' -n 2 entry && (( ${#entry[@]} != 0 )); do
+            type=${entry[0]}
+            file=${entry[1]}
+            case $type in
+              A*)
+                source=""
+                dest=$file
+                ;;
+              M*)
+                source=$file
+                dest=$file
+                ;;
+              C*|R*)
+                source=$file
+                read -r -d '' dest
+                ;;
+              *)
+                echo "Ignoring file $file with type $type"
+                continue
+            esac
+
+            if [[ -n "$source" ]] && [[ "$(nixf-tidy --variable-lookup < ${{ env.base }}/"$source")" != '[]' ]] 2>/dev/null; then
+              echo "Ignoring file $file because it doesn't pass nixf-tidy in the base commit"
+              echo # insert blank line
+            else
+              nixf_report="$(nixf-tidy --variable-lookup < "$source")"
+              if [[ "$nixf_report" != '[]' ]]; then
+                echo "$file doesn't pass nixf-tidy. Reported by nixf-tidy:"
+                echo # insert blank line
+                echo "$nixf_report" | jq -r '
+                  def getCur: "line " + (.line+1|tostring) + ", column " + (.column|tostring);
+                  def getRange: "[" + (.lCur|getCur) + " ~ " + (.rCur|getCur) + "]";
+                  def getNote: . as $top|(.range|getRange) + " "+ .sname + ": " +
+                    (.message|sub("{}" ; ($top.args.[]|tostring)));
+                  def getNoteNewline: "\n  " + (.|getNote);
+                  def getMessage: (.|getNote) + (if (.notes|length)>0 then
+                    (.notes.[]|getNoteNewline) else "" end) + "\n";
+                  .[]|getMessage
+                ' | sed 's/^/  /' # indent 2 spaces
+                failedFiles+=("$file")
```

```suggestion
                failedFiles+=("$dest")
```

This should also be fixed in https://github.com/NixOS/nixpkgs/blob/eecca18adfeaf293a151a11003f70c8d8b6cb380/.github/workflows/check-nix-format.yml#L79 my mistake